### ISSUE TITLE
comment when encountering an error that is user-resolvable

### ISSUE THREAD

**Issue by jmcampanini** [palantir/bulldozer#63]
there are times when bulldozer encounters an error that is user-resolvable, specifically when a merge fails due to permission settings.

we currently just log these, example:

```
Merge rejected due to unsatisfied condition \"You're not authorized to push to this branch. Visit https://help.github.com/enterprise/2.15/user/articles/about-protected-branches/ for more information.\
```

in this particular case, we should be able to add a comment to the PR with feedback on what happened and possible resolution steps

**Comment by jmcampanini** [comment]
* 2: `X of Y required status checks have not succeeded: Z expected and A pending.`
* 39: `Base branch was modified. Review and try the merge again.`
* 2: `Merge commits are not allowed on this repository.`
* 63: `You're not authorized to push to this branch. Visit https://help.github.com/enterprise/2.15/user/articles/about-protected-branches/ for more information.`

**Comment by jmcampanini** [comment]
seems like the one we should focus on surfacing to the user is:

> You're not authorized to push to this branch...

**Comment by bluekeyes** [comment]
Might want to include `Merge commits are not allowed on this repository.` too as I think that implies that the Bulldozer configuration does not match the repository configuration.

**Comment by jmcampanini** [comment]
closed by #67
